controllers/flinkcluster_submit_job_script.go

```diff
@@ -70,7 +70,7 @@ function wait_for_job() {
 		# Find active job first.
 		# If the current job is restarted by the operator, there will be records of past stopped jobs.
 		# TODO: It needs to be improved to determine the job state with the submitted job id.
-		if list_jobs | grep -e "(SCHEDULED)" -e "(RUNNING)"; then
+		if list_jobs | grep -e "(SCHEDULED)" -e "(RUNNING)" -e "(RESTARTING)"; then
```

Comment:
Could you add check for all states defined [here](https://ci.apache.org/projects/flink/flink-docs-release-1.9/api/java/org/apache/flink/runtime/jobgraph/JobStatus.html)?  Treat `CREATED`, `CANCELLING`, `FAILING`, `RECONCILING`, `SUSPENDED` as active.

Reply:
Also, for unknown job state, don't exit, it will eventually reach terminal state.

Reply:
Added more statuses to treat as active jobs.

Reply:
https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/bf30b7b88941b8491ddcaa2935bfa4c97a409784/controllers/flinkcluster_submit_job_script.go#L52
Do I need to add some status here as well?

Reply:
I mean for line 89, don't exit for unknown state, it will retry later.

Reply:
All right. 👍

Reply:
Right, you need to add them to line 52 as well.

Reply:
I added it there as well. I added a line break because it was long on the side.